tasks/insertionTasks.ts

```diff
@@ -0,0 +1,260 @@
+
+        // If PR was created and we're not in dry run mode, update the changelog with the PR number
+        if (prNumber && !options.dryRun) {
+            console.log(`PR #${prNumber} created. Updating changelog with PR link...`);
+
+            // Update changelog with PR number (single call)
+            await updateChangelog(options.roslynVersion, prList, prNumber);
+
+            // Create a second commit to include the updated changelog and push to update the PR
+            await git(['add', 'CHANGELOG.md']);
+            await git(['commit', '-m', `Update changelog with PR #${prNumber}`]);
+            await git(['push', 'target', `insertion/${options.roslynEndSHA}`]);
+
+            console.log(`Changelog updated with PR #${prNumber} link.`);
+        }
+
+    } catch (error) {
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        logError(`Insertion failed: ${errorMessage}`);
+        if (error instanceof Error && error.stack) {
+            console.log(`##[debug]${error.stack}`);
+        }
+        throw error;
+    }
+});
+
+async function extractRoslynVersionFromManifest(manifestPath: string): Promise<string | null> {
+    const xmlFile = path.join(manifestPath, 'OfficialBuild.xml');
+
+    if (!fs.existsSync(xmlFile)) {
+        logError(`OfficialBuild.xml not found at ${xmlFile}`);
+        return null;
+    }
+
+    const xmlContent = fs.readFileSync(xmlFile, 'utf8');
+    const parser = new xml2js.Parser();
+    const result = await parser.parseStringPromise(xmlContent);
+
+    const packages = result?.Build?.Package || [];
+    for (const pkg of packages) {
+        const attrs = pkg.$;
+        if (attrs?.Id === 'Microsoft.CodeAnalysis.Common') {
+            return attrs.Version;
+        }
+    }
+
+    logError('Microsoft.CodeAnalysis.Common package not found in the asset manifest.');
+    return null;
+}
+
+async function verifyRoslynRepo(roslynRepoPath: string): Promise<void> {
+    if (!fs.existsSync(roslynRepoPath)) {
+        throw new Error(`Roslyn repository not found at ${roslynRepoPath}`);
+    }
+    console.log(`Using Roslyn repository at ${roslynRepoPath}`);
+}
+
+async function generatePRList(startSHA: string, endSHA: string, roslynRepoPath: string, _options: InsertionOptions): Promise<string | null> {
+    console.log(`Generating PR list from ${startSHA} to ${endSHA}...`);
+
+    try {
+        const { stdout } = await execAsync(
+            `cd "${roslynRepoPath}" && roslyn-tools pr-finder -s "${startSHA}" -e "${endSHA}" --format changelog --label VSCode`,
+            { maxBuffer: 10 * 1024 * 1024 } // 10MB buffer
+        );
+        return stdout && stdout.trim() ? stdout : null;
+    } catch (error) {
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        logWarning(`PR finder failed: ${errorMessage}`);
+        if (error instanceof Error && error.stack) {
+            console.log(`##[debug]${error.stack}`);
+        }
```

Comment:
Similar comment to the above, consider placing some of these error message and error stack trace determinations in the `logError`/`logwarning` functions